### DISCUSSION TITLE
Read10X applied on a very large dataset

### DISCUSSION THREAD

**Discussion by indraroy** [satijalab/seurat#5575] [Q&A] [answered]
I am very new to this and am stuck on a pretty elementary step. I am using a large public dataset containing scRNA seq data and the mtx.gz file is about 4.8GB. With the features and barcodes files they are almost 5GB. 
I am having issues reading in the data and keep getting the following error
                    > Error: cannot allocate vector of size 6.2 Gb

Here is the chunk.
`
pathway <- "C:/.../PDHomo data/"
list.files(pathway)
PDHomo.data <- Read10X(data.dir = pathway)
`

It's been suggested that I am missing some package or dependencies but when I tried applying this to a different, much smaller dataset (100mb) it worked just fine. 

Is there a certain way of reading in a very large matrix? What am I missing here, please help?

**Comment by samuel-marsh** [comment]
Hi,

How much RAM do you have on the system that you are running this on?

Best,
Sam

**Reply by indraroy** [reply]
I have tried it on device with 8GB and another with 16GB.

**Reply by samuel-marsh** [reply]
The error you are getting is result of not having enough memory.  I would try running on system with larger amount of available memory.

Best,
Sam

**Comment by indraroy** [comment, marked as answer]
As suggested here & elsewhere, the problem was in the available RAM size of my device(s).
